tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -270,7 +318,21 @@ final class ScalaJSClassEmitter(
       val js.Function(params, body) = genConstructorFunForJSClass(tree)
       js.MethodDef(static = false, js.Ident("constructor"), params, body)
     } else {
+      val className = tree.name.name
       val fieldDefs = genFieldDefs(tree)
+
+      val (otherStatements: js.Tree, args) = if (usesJSConstructorOpt(className)) {
```

Comment:
Remove `: js.Tree`, it's bad for type safety (it's a pattern matching in that position, not a type ascription)